flink-python/pyflink/datastream/tests/test_window.py

```diff
@@ -441,6 +442,35 @@ def process(self, key, context: ProcessWindowFunction.Context,
         expected = ['(hi,2)', '(hi,2)', '(hi,2)']
         self.assert_equals_sorted(expected, results)
 
+    def test_event_time_tumbling_window_all(self):
+        data_stream = self.env.from_collection([
+            ('hi', 1), ('hello', 2), ('hi', 3), ('hello', 4), ('hello', 5), ('hi', 8), ('hi', 9),
+            ('hi', 15)],
+            type_info=Types.TUPLE([Types.STRING(), Types.INT()]))  # type: DataStream
+        watermark_strategy = WatermarkStrategy.for_monotonous_timestamps() \
+            .with_timestamp_assigner(SecondColumnTimestampAssigner())
+
+        class CountAllWindowProcessFunction(ProcessAllWindowFunction[tuple, tuple, TimeWindow]):
+            def process(self, context: 'ProcessAllWindowFunction.Context',
+                        elements: Iterable[tuple]) -> Iterable[tuple]:
+                return [
+                    (context.window().start, context.window().end, len([e for e in elements]))]
+
+            def clear(self, context: 'ProcessAllWindowFunction.Context') -> None:
+                pass
+
+        data_stream.assign_timestamps_and_watermarks(watermark_strategy) \
+            .key_by(lambda x: x[0], key_type=Types.STRING()) \
```

Comment:
window_all should applied to DataStream instead of KeyedStream and so please remove this line.